### ISSUE TITLE
CI4: Problem in Module model

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#3349]
### Background information

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.3.5

### Issue / Bug  / Question / New Feature

The Following code appears at the end of the Module model and I need to properly convert it to the CI4 equivalent.  The problem is that the query doesn't make sense to me.  This code was added about four years ago in code that @daN4cat and @SteveIreland worked on to add the office menu.

```
	/**
	 * This method is used to show the office navigation icon on the home page
	 * which happens when the sort value is greater than zero
	 */
	public function get_show_office_group()
	{
		$this->db->select('sort');
		$this->db->from('grants');
		$this->db->where('module_id', 'office');
		$this->db->from('modules');
		return $this->db->get()->row()->sort;
	}
```

The function seems to be returning the value of sort (int) from the modules table.  Both the sort and module_id fields appear in the modules table.  The question is: what is the intention with the `$this->db->from('grants');` line? Was it meant to be a `JOIN modules ON grants.menu_group = modules.module_id`?  

I may be mistaken, but I've never seen multiple from calls in the same QueryBuilder calls in the same way that you can do multiple where() or join().  Can anyone shed some light on this one for me?

**Comment by daN4cat** [comment]
My mind is blank on this story I'm afraid.

**Comment by jekkos** [comment]
Not sure how this query looks like, can you print a `this->db->last_query()` after to see how it  looks like?

**Comment by SteveIreland** [comment]
Looks like a copy and paste that went bad.  I believe that the intent was to replace grants with modules.  It wasn't caught in testing because apparently the generated SQL does enough that it  passed the limited testing that was done on it.

**Comment by objecttothis** [comment]
> Looks like a copy and paste that went bad. I believe that the intent was to replace grants with modules. It wasn't caught in testing because apparently the generated SQL does enough that it passed the limited testing that was done on it.

Thank you @SteveIreland. If I understand correctly, the code should look like this?

```
/**
	 * This method is used to show the office navigation icon on the home page
	 * which happens when the sort value is greater than zero
	 */
	public function get_show_office_group()
	{
		$this->db->select('sort');
		$this->db->where('module_id', 'office');
		$this->db->from('modules');

		return $this->db->get()->row()->sort;
	}
```
This makes sense because the grants table do not have a columns called sort or module_id.  We should probably test that change outside of my change from CI3 to CI4.  God knows that the CI3->CI4 change is going to be heavily regression tested.

**Comment by SteveIreland** [comment]
That should work.

**Comment by mrsarith** [comment]
Hello,

May you help me,please? I have one issue relate missing title on Modal form. I dont know I touch sth where else. That why, all title got blank.

here is my screenshot.

https://docs.google.com/document/d/1mSDYP13qqObWKmM7xpc55W_YmVk7Nn6T9-CJq0nO8pE/edit?usp=sharing

]
Sarith

**Comment by daN4cat** [comment]
@mrsarith , is that from a modified OSPOS?

**Comment by objecttothis** [comment]
Fixed in #3271
